ui/app/store/actions.js

```diff
@@ -1178,15 +1178,15 @@ export function setSelectedToken (tokenAddress) {
 }
 
 export function setSelectedAddress (address) {
-  return (dispatch) => {
+  return async (dispatch) => {
     dispatch(showLoadingIndication())
     log.debug(`background.setSelectedAddress`)
-    background.setSelectedAddress(address, (err) => {
+    try {
+      await promisifiedBackground.setSelectedAddress(address)
+    } catch (error) {
       dispatch(hideLoadingIndication())
```

Comment:
I think we might need to move this out of the catch block, or it won't ever be hidden?

Reply:
A similar case exists in #8427

Reply:
🤦‍♂️ Thanks, good catch. I don't know how I missed this. I'll update both PRs.

Reply:
Updated. I believe #8427 already calls `hideLoadingIndication()` outside of the `catch` block. I duplicated it instead of putting it in a `finally` block to preserve the existing behavior as much as possible, including the order in which `hideLoadingIndication` and `displayWarning` are called in the failure case.

Reply:
Can we add a `return` statement at the end of the `catch` block here, to prevent two dispatches of this

Reply:
Similar to #8427

Reply:
Welp. Updated. Clearly this is a sign that I need to review these more carefully. This is what I get for trying to multitask.

Hopefully it'll finally work this iteration! 😬 😅